### PR TITLE
BF: DOCS: fix slash for windows in conf.py

### PR DESCRIPTION
**Pull request by brunobeltran**
## PR Summary

Single line fix to remove a hard-coded slash that was preventing the docs from building on windows in Anaconda.

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -158,7 +159,7 @@ def _check_dependencies():
         'numpy': 'https://docs.scipy.org/doc/numpy/',
         'scipy': 'https://docs.scipy.org/doc/scipy/reference/',
     },
-    'backreferences_dir': 'api/_as_gen',
+    'backreferences_dir': Path('api') / Path('_as_gen'),
```
Comment (tacaswell) [inline doc/conf.py:162]: as a side-comment, you don't actually need the second `Path` as they define "division" on include str on the right side.

**Review by tacaswell — APPROVED** [review]
Anyone can merge on green.

File: doc/conf.py (hunk 1 above)
Comment (anntzer) [reply doc/conf.py:162]: You can also write `Path('api', '_as_gen')` or (my preference) `Path('api/_as_gen')` (pathlib will normalize the slash to a backslash for you).

**Comment by brunobeltran** [comment]
Some extension seems to have changed under us? Doc build works locally, but is failing on CI.

**Comment by jklymak** [comment]
rebase - fix is on master

**Review by jklymak — APPROVED** [review]

**Pull request merged**